pkg/autoscaler/statforwarder/forwarder.go

```diff
@@ -130,6 +136,18 @@ func New(ctx context.Context, logger *zap.SugaredLogger, kc kubernetes.Interface
 	return f
 }
 
+func (f *Forwarder) extractIP(holder string) (string, error) {
+	if ip, ok := f.id2ip[holder]; ok {
+		return ip, nil
+	}
+
+	_, ip, err := bucket.ExtractPodNameAndIP(holder)
+	if err == nil {
+		f.id2ip[holder] = ip
+	}
+	return ip, err
```

Comment:
"", err?

Reply:
err could be nil here then ip should be returned.